src/main/java/org/takes/rq/RqSocket.java

```
@@ -0,0 +1,119 @@
+        super(req);
+    }
+
+    /**
+     * Returns IP address from the X-Takes-LocalAddress header.
+     * @return Local InetAddress
+     * @throws IOException If fails
+     */
+    public InetAddress getLocalAddress() throws IOException {
+        return InetAddress.getByName(this.parseHeader("X-Takes-LocalAddress"));
+    }
+
+    /**
+     * Returns IP address from the X-Takes-RemoteAddress header.
+     * @return Remote InetAddress
+     * @throws IOException If fails
+     */
+    public InetAddress getRemoteAddress() throws IOException {
+        return InetAddress.getByName(this.parseHeader("X-Takes-RemoteAddress"));
+    }
+
+    /**
+     * Returns port from the X-Takes-LocalPort header.
+     * @return Local Port
+     * @throws IOException If fails
+     */
+    public int getLocalPort() throws IOException {
+        return this.getPort("X-Takes-LocalPort");
+    }
+
+    /**
+     * Returns port from the X-Takes-RemotePort header.
+     * @return Remote Port
+     * @throws IOException If fails
+     */
+    public int getRemotePort() throws IOException {
+        return this.getPort("X-Takes-RemotePort");
+    }
+
+    /**
+     * Parses value of the provided header from head.
+     * @param header Header to parse
+     * @return String value of the provided header
+     * @throws IOException If fails
+     */
+    private String parseHeader(final CharSequence header) throws IOException {
+        String result = "";
+        for (final String line : this.head()) {
+            final String[] parts = line.split(": ", 2);
+            if (parts[0].contains(header)) {
+                result = parts[1];
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Parses port value from the provided header.
+     * @param header Header to parse
+     * @return Integer value of the port
+     * @throws IOException If fails
+     */
+    private int getPort(final CharSequence header) throws IOException {
```

@ikhvostenkov I don't see the value of this method exactly. Why are we defaulting to zero if there is no header value? It should throw an exception just like in other cases. Let's just call `Integer.parseInt(this.parseHeader(header))`.